com.unity.netcode.gameobjects/Runtime/Messaging/MessagingSystem.cs

```diff
@@ -75,6 +78,35 @@ internal struct MessageWithHandler
             public MessageHandler Handler;
         }
 
+        internal List<MessageWithHandler> PrioritizeMessageOrder(List<MessageWithHandler> allowedTypes)
```

Comment:
Your reorder method can break this assumption, can it not?

Reply:
I suppose I'm OK with the approach, but I think it's worth the time to add more checks & safe-guards so that we cannot easily break those assumptions in the future.

e.g. If there's an assumption that these 3 message types will always be the first three, that means no one should ever reorder the first 3 indices, so we should throw or assert if it happens. This type of thing will help future us maintain the thing since the logic of this thing is fairly distributed in multiple files & methods.